nncf/layer_utils.py

```diff
@@ -79,3 +79,21 @@ def forward(self, *args):
             if op_results is not None:
                 results = op_results
         return results
+
+
+class CompressionParameter(nn.Parameter):
+    '''
+    The class that should be used in all compression algorithms instead of torch.nn.Parameter.
+
+    This class utilize `compression_lr_multiplier` parameter from :class:`nncf.NNCFConfig`
+    to increase/decrease gradients for compression algorithms' parameters.
+    '''
+
+    def __new__(cls, data=None, requires_grad=True, compression_lr_multiplier=None):
+        return super().__new__(cls, data, requires_grad=requires_grad)
+
+    def __init__(self, data=None, requires_grad=True, compression_lr_multiplier=None):
+        super().__init__()
+        if compression_lr_multiplier is not None and requires_grad:
+            self.register_hook(lambda grad: compression_lr_multiplier * grad)
```

Comment:
Set up type hints and add a docstring for `__init__`

Reply:
Done.